components/WorkSectionAsGallery.js

```diff
@@ -0,0 +1,221 @@
+import React, { PureComponent } from 'react';
+
+import Link from 'next/link';
+
+import PropTypes from 'prop-types';
+
+import { ContentfulMedia, SimpleFragment } from 'models';
+
+import { Markdown, Image } from 'components/base';
+
+import get from 'utils/get';
+import flattenImageData from 'utils/flattenImageData';
+
+const pauseStartVideo = (videoId) => {
+  const video = document.getElementById(videoId);
+  const videoButtonText = document.getElementById(`${videoId}--video-button-text`).textContent;
+
+  if (videoButtonText == '(Pause)') {
+    video.pause();
+    document.getElementById(`${videoId}--video-button-text`).textContent = '(Play)';
+  } else {
+    video.play();
+    document.getElementById(`${videoId}--video-button-text`).textContent = '(Pause)';
+  }
+}
+
+const renderWorkOverlay = (work) => {
+  const workIsImage = work.fields.asset.fields.file.contentType.startsWith("image/");
+
+  return (
+    <div className="WorkSectionAsGallery__work-hover-overlay">
+      <div className="WorkSectionAsGallery__work-hover-overlay--info">
+        <div className="WorkSectionAsGallery__work-hover-overlay--title">{work.fields.title}</div>
+        {work.fields.caseStudySlug ? 
+          <Link
+            href={work.fields.caseStudySlug}
+            passHref
+          >
+            <a
+              className="decoration-none color-white"
+              aria-label="read the case study"
+              rel="noopener noreferrer"
+            >
+            (read case study)
+            </a>
+          </Link> : 
+          <Link
+            href={work.fields.link}
+          >
+            <a
+              className="decoration-none color-white"
+              aria-label="visit the site"
+              target="_blank"
+              rel="noopener noreferrer"
+            >
+            → visit the site 
+            </a>
+          </Link>
+        }
+      </div>
+      {!workIsImage &&
+        <button 
+          id={`${work.sys.id}--video-button-text`}
+          className="WorkSectionAsGallery__work-hover-overlay--video-button"
+          onClick={() => pauseStartVideo(work.sys.id)}
+        >
+          (Pause)
+        </button>
+      }
+    </div>
+  );
+}
+
+const renderWorkAsImageOrVideo = (work) => {
+  const workIsImage = work.fields.asset.fields.file.contentType.startsWith("image/");
+  const workImage = flattenImageData(work.fields.asset);
+
+  return (
+    <div>
+      {workIsImage ? 
+        <Image
+          src={workImage.url}
+          width={workImage.width}
+          height={workImage.height}
+          sizes='50vw'
+        />  :
+        <video
+          id={work.sys.id}
+          key={work.sys.id}
+          style={{
+            width: '100%', 
+            height: '100%' 
+          }}
```

Comment:
we can use basement classnames here instead of inline styles `w100 h100`. inline styles should be reserved for cases where we need to pass in a value to the css, like `height: ${elemHeight}px`, or as last resort since it's the ultimate override

Reply:
fax